configs/components/puppet.rb

```diff
@@ -59,6 +63,83 @@
     pkg.install_service "ext/solaris/smf/puppet.xml", "ext/solaris/smf/puppet", service_type: "network"
   when "aix"
     pkg.install_service "resources/aix/puppet.service", nil, "puppet"
+  when "windows"
+    description = "Periodically fetches and applies configurations from a Puppet master server."
+    service_hash = [
```

Comment:
@Iristyle We did consider the alternative approach - See https://github.com/puppetlabs/puppet-agent/pull/514/files. It sounds like you think this might be safer.